erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -915,6 +920,8 @@ def make_internal_transfer_gl_entries(self, gl_entries):
 
 	def make_item_gl_entries(self, gl_entries):
 		# income account gl entries
+		enable_discount_accounting = cint(frappe.db.get_single_value('Accounts Settings', 'enable_discount_accounting'))
```

Comment:
Instead of fetching enable_discount_accounting twice, you can make it a class property.

Reply:
Done.